UnityProject/Assets/Scripts/Shuttles/CargoShuttle.cs

```diff
@@ -138,18 +141,21 @@ private Vector3 GetRandomFreePos()
 	{
 		int width = 2;
 		int height = 4;
+		int i = 0;
 
 		Vector3Int spawnPos;
-		while (true)
+		//temporary max crates in one
+		while (i < 40)
 		{
 			spawnPos = Vector3Int.RoundToInt(transform.position);
 			spawnPos.x += Random.Range(-width, width);
 			spawnPos.y += Random.Range(-height, height) + 1;
 			if (MatrixManager.Instance.GetFirst<ClosetControl>(spawnPos, true) == null)
 			{
-				break;
+				return spawnPos;
 			}
+			i++;
 		}
-		return spawnPos;
+		return Vector3.zero;
```

Comment:
use `TransformState.InvalidPos` as null object, because `Vector3.zero` can be valid spawning position

Reply:
You mean `TransformState.HiddenPos`?

Reply:
yeah, HiddenPos